pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -146,25 +139,11 @@ let
     (optionalString (versionOlder "7.10" ghc.version && !isHaLVM) "-threaded")
   ];
 
-  isHaskellPkg = x: (x ? pname) && (x ? version) && (x ? env);
-  isSystemPkg = x: !isHaskellPkg x;
-
-  allPkgconfigDepends = pkgconfigDepends ++ libraryPkgconfigDepends ++ executablePkgconfigDepends ++
-                        optionals doCheck testPkgconfigDepends ++ optionals doBenchmark benchmarkPkgconfigDepends;
+  inherit (haskellLib.extractBuildInputs ghc (args // { inherit doCheck doBenchmark; }))
+    allPkgconfigDepends propagatedBuildInputs otherBuildInputs haskellBuildInputs systemBuildInputs;
 
   nativeBuildInputs = optional (allPkgconfigDepends != []) pkgconfig ++
```

Comment:
Don't forget these too!

Reply:
Nothing has changed with these, they're not available in pkg.env either...

Reply:
@shlevy What do you mean nothing has changed?

As to them not being in `pkg.env`, that's an error I have some disorganized commits I intend to add to #26799 after #26805 finally lands. I started making a `buildInputs` vs `nativeBuildInputs` distinction in https://github.com/NixOS/nixpkgs/pull/26480/files and should have fixed `pkg.env` then.

Reply:
I just mean that the definition of nativeBuildInputs hasn't changed nor has its usage. Nothing in this PR required extracting it out. There's no reason not to do so in the future if there's a need, but there's also no reason to do so now...

Reply:
That's fine then.